doc/advanced_topics.md

```diff
@@ -86,6 +86,120 @@ The Readers and Writers that ship with podio are assumed to run on a single
 synchronize with any other for file operations).
 
+
+## Schema evolution
+
+The podio toolkit supports schema evolution for EDMs that are generated with it.
+it also tries to leverage existing schema evolution functionality of the chosen
+backend if available, e.g. via ROOT. In order to opt-in to schema evolution, it
+is necessary to pass the old datamodel definition(s) to the code generator, via:
+
+::::{tab-set}
+:::{tab-item} CMake macro
+
+```cmake
+PODIO_GENERATE_DATAMODEL(datamodel datamodel.yaml headers sources
+    OLD_DESCRIPTION old_datamodel.yaml
+)
+```
+
+:::
+:::{tab-item} python CLI script
+
+```bash
+python3 python/podio_class_generator.py \
+    --old-description old_datamodel.yaml \
+    datamodel.yaml output_gen_dir datamodel
+```
+
+:::
+::::
+
+In case podio detects unsupported schema changes code generation will fail with
+a message about which changes were detected that are not supported. For some
+evolutions we have implemented support for manually specifying what should
+happen. In these cases it is necessary to specify an *evolution file* that
+describes these changes, the grammar is as follows
+
+```yaml
+migrations:
+  <datatype>:  # The datatype (or component) for which a change happens
+    - from_version: <version>  # The old schema version (before the change)
+      to_version: <version>  # The new schema version (after the change)
+      <change_type>: {<change_args>}
+```
+
+It's possible to specify multiple changes for any given datatype and the
+`{from,to}_version` as well as the `<change_type>` are mandatory.
+
+::::{tab-set}
+:::{tab-item} CMake macro
+
+```cmake
+PODIO_GENERATE_DATAMODEL(datamodel datamodel.yaml headers sources
+    OLD_DESCRIPTION old_datamodel.yaml
+    SCHEMA_EVOLUTION evolution.yaml
+)
+```
+
+:::
+:::{tab-item} python CLI script
+
+```bash
+python3 python/podio_class_generator.py \
+    --old-description old_datamodel.yaml \
+    --evolution_file evolution.yaml \
+    datamodel.yaml output_gen_dir datamodel
+```
+
+:::
+::::
+
+### Supported manual evolutions
+
+| `<change_type>` | description              | `<change_args>`                      |
+|-----------------|--------------------------|--------------------------------------|
+| `rename_member` | Rename a member variable | `{from: <old-name>, to: <new-name>}` |
+
+An example evolution file looks like
+```yaml
+migrations:
+  TestType:
+    - from_version: 1
+      to_version: 2
+      rename_member:
+        from: oldEnergy
+        to: energy
+```
+
+### Checking possiblities for schema evolution
+
+The podio toolkit also provides a small script that allows to check up-front
+whether two versions of an EDM need manual intervention as well as confirming
+that an evolution file indeed handles all changes that require manual
+intervention. This can be called as
+
+```bash
+python3 python/podio_schema_evolution.py \
+  datamodel.yaml old_datamodel.yaml
```

Comment:
Maybe this should go from older to newer? That is the order followed in every other place in the new documentation.